protocol/x/affiliates/keeper/keeper.go

```diff
@@ -259,6 +267,52 @@ func (k Keeper) GetIndexerEventManager() indexer_manager.IndexerEventManager {
 	return k.indexerEventManager
 }
 
+func (k Keeper) GetAffiliateWhitelistMap(ctx sdk.Context) (map[string]uint32, error) {
+	affiliateWhitelist, err := k.GetAffiliateWhitelist(ctx)
+	if err != nil {
+		return nil, err
+	}
+	affiliateWhitelistMap := make(map[string]uint32)
+	for _, tier := range affiliateWhitelist.GetTiers() {
+		for _, address := range tier.GetAddresses() {
+			affiliateWhitelistMap[address] = tier.GetTakerFeeSharePpm()
+		}
+	}
+	return affiliateWhitelistMap, nil
+}
+
+func (k Keeper) SetAffiliateWhitelist(ctx sdk.Context, whitelist types.AffiliateWhitelist) error {
+	store := ctx.KVStore(k.storeKey)
+	addressSet := make(map[string]bool)
+	for _, tier := range whitelist.Tiers {
```

Comment:
New change: let's also add a protocol constant = 50% as upper bound of the fee share ppm, for fee security

Reply:
I think it would be better to add this to `ValidateRevShareSafety`


Now that I think about it, Im gonna make `ValidateRevShareSafety` more accurate - can discuss offline

Reply:
To clarify, we explicitly want a constant `AffiliateWhitelistCap = 50%` for the affiliate whitelist fee share - don't see being added?